src/components/Dashboard/scenes/ManageMentorApplications/ManageMentorApplications.tsx

```diff
@@ -1,11 +1,202 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { type Mentor } from '../../../../types';
+import { useMentors } from '../../../../hooks/useMentors';
+import useCategories from '../../../../hooks/useCategories';
+import { useQueryClient } from '@tanstack/react-query';
+
+export const ManageMentorApplications: React.FC = () => {
+  const [filter, setFilter] = useState('');
+  const [searchTerm, setSearchTerm] = useState('');
+  const { data: categories } = useCategories();
+  const [categoryFilter, setCategoryFilter] = useState<string>('');
+  const queryClient = useQueryClient();
+  const { isLoading, data: mentors, updateStatus } = useMentors(categoryFilter);
+
+  const handleStatusUpdate = async (mentorId: string, newStatus: string) => {
+    try {
+      await updateStatus({ mentorId, newStatus });
+    } catch (error) {
+      console.error('Error updating mentor status:', error);
+    }
+  };
+
+  const handleCategoryChange = async (
+    e: React.ChangeEvent<HTMLSelectElement>
+  ) => {
+    const selectedCategory = e.target.value;
+    setCategoryFilter(selectedCategory);
+    await queryClient.refetchQueries({
+      queryKey: ['mentors', selectedCategory],
+    });
+  };
+
+  const renderFilters = () => {
+    const filters = [
+      { label: 'All', status: '' },
+      { label: 'Approved', status: 'approved' },
+      { label: 'Pending', status: 'pending' },
+      { label: 'Rejected', status: 'rejected' },
+    ];
+
+    return (
+      <>
+        {mentors !== undefined && (
+          <div className="flex mb-3">
+            {filters.map(({ label, status }) => {
+              const count = mentors.filter((mentor) =>
+                status.length > 0 ? mentor.state === status : true
+              ).length;
+              return (
+                <button
+                  key={label}
+                  className={`px-4 py-1 font-medium  ${
+                    filter === status
+                      ? 'bg-blue-500 text-white'
+                      : 'bg-gray-200 text-gray-700 hover:bg-gray-300'
+                  } ${
+                    label === 'All'
+                      ? 'rounded-l-md'
+                      : label === 'Rejected'
+                      ? 'rounded-r-md'
+                      : 'rounded-none'
+                  }`}
+                  onClick={() => {
+                    setFilter(status);
+                  }}
+                >
+                  {label} ({count})
+                </button>
+              );
+            })}
+          </div>
+        )}
+      </>
+    );
+  };
+
+  const renderMentorTable = () => {
+    const filteredMentors =
+      mentors != null
+        ? filter !== ''
+          ? mentors.filter((mentor) => mentor.state === filter)
+          : mentors
+        : [];
```

Comment:
why can't we use this like this
`const filteredMentors = mentors?.filter(mentor => mentor.state === filter) || [];`